sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -366,7 +368,7 @@ def _process_base_ops(op: BaseOp):
     return op
 
 
-def _op_to_template(op: BaseOp, enable_artifacts=False):
+def _op_to_template(op: BaseOp, tekton_compiler):
```

Comment:
All the op-to-template conversions here are static in a sense that they do not rely on the compiler instance. Any compiler instance specific template manipulations are in the `TektonCompiler` class. Can the output artifact tracking annotation code move to the `TektonCompiler`, or, alternatively we pass the `output_artifacts` into this `_op_to_template` method?

Reply:
I changed the arguments to only pass the necessary objects. I also added a new test case because we are going to remove the `artifact_location` test after Feng's PR. (cc #163)